pkg/kinflate/resource/appresource.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resource
+
+import (
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	kutil "k8s.io/kubectl/pkg/kinflate/util"
+	"k8s.io/kubectl/pkg/kinflate/util/fs"
+)
+
+type appResource struct {
+	data *unstructured.Unstructured
+}
+
+func (ap *appResource) Name() string {
+	if ap.data == nil {
+		return ""
+	}
+	metaAccessor, err := meta.Accessor(ap.data)
+	if err != nil {
+		return ""
+	}
+	return metaAccessor.GetName()
+}
+
+func (ap *appResource) Data() unstructured.Unstructured {
+	return *ap.data
+}
+
+type appResourceList struct {
+	path string
+	fsys fs.FileSystem
+}
+
+func NewAppResourceList(path string) *appResourceList {
```

Comment:
Yeah, eventually will take the loader and the path

Reply:
Added loader to the function parameters. Now the function is `ResourcesFromPath(path string, loader Loader)`